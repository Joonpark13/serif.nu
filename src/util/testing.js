/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import * as reactRedux from 'react-redux';
import { shallow } from 'enzyme';
import * as useSelector from './use-selector';

jest.mock('./use-selector');

// Helper function that returns a function which creates enzyme wrappers with default props
export function wrapperCreator(Component, defaultProps = {}) {
  let classes;
  return props => shallow(<Component classes={classes} {...defaultProps} {...props} />);
}

/* istanbul ignore next */
export function mockUseSelector(...args) {
  useSelector.default.mockReset();
  args.forEach(arg => useSelector.default.mockReturnValueOnce(arg));

  return useSelector.default;
}

/* istanbul ignore next */
export function mockUseDispatch() {
  const dispatchSpy = jest.fn();
  reactRedux.useDispatch.mockReturnValue(dispatchSpy);
  return dispatchSpy;
}
