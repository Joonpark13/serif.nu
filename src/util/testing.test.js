import React from 'react';
import { mockStyles, wrapperCreator } from './testing';

describe('Testing Utils', () => {
  describe('mockStyles', () => {
    it('returns a object with each style property as empty strings', () => {
      const style = {
        icon: {
          color: 'white',
        },
        title: {
          marginBotton: 2,
        },
      };

      expect(mockStyles(style)).toEqual({ icon: '', title: '' });
    });
  });

  describe('wrapperCreator', () => {
    it('creates a wrapper creator function', () => {
      const TestComponent = () => <div />;
      const wrapper = wrapperCreator(TestComponent);
      expect(wrapper().get(0)).toMatchSnapshot();
    });
  });
});
