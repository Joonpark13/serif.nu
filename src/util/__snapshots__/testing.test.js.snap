// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Utils wrapperCreator creates a wrapper creator function 1`] = `<div />`;
